src/cmd-run

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+# Forked from https://github.com/coreos/scripts/blob/master/build_library/qemu_template.sh
+# Changed to have command line arguments, drop non-x86_64/non-KVM support
+# Automatically uses `-snapshot` if the target disk isn't writable
+# Uses -nographic by default, and most importantly, contains a default
+# Ignition config that auto-logins on the console
+
+set -euo pipefail
+
+VM_DISK=
+VM_MEMORY=2048
+VM_PERSIST=0
+VM_NCPUS="${VM_NCPUS:-$(nproc)}"
+VM_SRV_MNT=
+SSH_PORT=${SSH_PORT:-}
+USAGE="Usage: $0 /path/to/disk.qcow2 [--] [qemu options...]
+Options:
+    -d DISK     Root disk drive (won't be changed by default)
+    --persist   Don't create a temporary snapshot
+    -i FILE     File containing an Ignition config
+    --srv src   Mount (via 9p) src on the host as /var/srv in guest
+    -m MB       RAM size in MB (2048)
+    -p PORT     The port on localhost to map to the VM's sshd. [2222]
+    -h          this ;-)
+
+This script is a wrapper around qemu for starting CoreOS virtual machines,
+it will auto-log you into the console, and by default for read-only disk
+images makes a transient snapshot.
+
+Any arguments after -a and -p will be passed through to qemu, -- may be
+used as an explicit separator. See the qemu(1) man page for more details.
+"
+
+die(){
+	echo "${1}" 1>&2
+	exit 1
+}
+
+while [ $# -ge 1 ]; do
+    case "$1" in
+        -d)
+            VM_DISK="$2"
+            shift 2 ;;
+        --persist)
+            VM_PERSIST=1
+            shift 1 ;;
+        -i|-ignition-config)
```

Comment:
`--ignition-config`? Looks like there's a mix of GNU and golang conventions here.